### ISSUE TITLE
Tethering is shown to access location every now and then, permission not requested nor granted

### ISSUE THREAD

**Issue by xoraxax** [GrapheneOS/os-issue-tracker#2178]
On my GrapheneOS pixel 6 phone, I am seeing "Tethering" performing location access (green dot in upper right corner), despite "Tethering" not having this permission requested nor granted. This happens quite a few times per day.

Is this a bug in the sandboxing layer? Is this an AOSP bug in the display of the requesting app?

**Comment by thestinger** [comment]
You need to say which kind of tethering you're using. It's not a bug. It should be on the list of exclusions for OS components. Location data includes data on nearby devices, etc.

**Comment by xoraxax** [comment]
I am not using any tethering.

**Comment by xoraxax** [comment]
Please reopen @thestinger if this information is useful.

**Comment by thestinger** [comment]
Which specific OS component is appearing as accessing location? Need the name and app id from the Settings app page for it.

**Comment by xoraxax** [comment]
This app settings page is shown when I click the "Tethering is accessing your location" line (shown after I click the green chip in the upper right corner) - naturally, this component of AOSP does not ask for location permissions according to the settings page.

![image](https://user-images.githubusercontent.com/1299820/236636806-f1cc00f6-578f-4d57-afd7-a1bb9505c286.png)
![image](https://user-images.githubusercontent.com/1299820/236636816-6c010fba-2813-43ee-9d3e-11e28de86d49.png)

Excuse the German locale.

**Comment by thestinger** [comment]
You don't have Bluetooth tethering or Wi-Fi hotspot enabled? Is Bluetooth itself enabled?

**Comment by xoraxax** [comment]
Bluetooth is enabled, bluetooth tethering is disabled and was never used. Wifi hotspot is configured but disabled. I can disable bluetooth to see if this changes the weird behaviour.

**Comment by xoraxax** [comment]
Maybe some apex patches the tethering module? My phone is not rooted, I cannot check.

**Comment by thestinger** [comment]
> Maybe some apex patches the tethering module? My phone is not rooted, I cannot check.

Unsure what you mean by that. Updates have to be signed both at update time and following that due to verified boot. We don't ship out-of-band APEX updates at the moment.

**Comment by xoraxax** [comment]
Lets assume I have a pristine /system partition, maybe the updating apex is picked up from /data/... or some other path?

I mean this behaviour is not explainable assuming that a) Tethering does not need location access most likely, b) Tethering is disabled, c) GOS Sandboxing should disallow effective location access, even for the Tethering apex.

**Comment by r3g-5z** [comment]
We still do not ship out of band APEX updates. You cannot just apply APEX updates yourself.

Furthermore, the NetworkStack APEX module has not had any changes in many months beyond AOSP translations and the app ID is not related to any modules. This has nothing to do with APEX modules.

**Comment by thestinger** [comment]
No, you're wrong. Read what I said about signing above.

Location doesn't specifically mean location services as I mentioned above. Tethering service has privileged Bluetooth, Tethering and other permissions. Some uses of Bluetooth are considered Location data access and it doesn't need to follow the usual rules because it's a privileged app.

**Comment by thestinger** [comment]
@r3g-5z It's part of the Connectivity module (packages/modules/Connectivity/Tethering).

**Comment by xoraxax** [comment]
Good point, but why would Tethering would request anything location related even when its not used/enabled?

**Comment by thestinger** [comment]
Bluetooth scans count as location access. It is almost certainly doing a Bluetooth scan or something else with Bluetooth considered to be location access. I expect it stops if you disable Bluetooth. The question would be why is it using that when you have Bluetooth tethering off.

**Comment by xoraxax** [comment]
Bluetooth disabled, Tethering and the green dot:

![image](https://user-images.githubusercontent.com/1299820/236638990-ea59653d-603e-49ea-8c39-26f2f7e14c4f.png)
![image](https://user-images.githubusercontent.com/1299820/236639000-97a2d410-8204-41f5-9fea-41c5ec79e01a.png)

**Comment by thestinger** [comment]
Try turning off Wi-Fi hotspot completely.

**Comment by xoraxax** [comment]
There seems to be no setting that is more disabled than the current state of being disabled:

![image](https://user-images.githubusercontent.com/1299820/236639225-cbbbc5e1-a413-47f6-9a52-ddbef5a32cff.png)

**Comment by thestinger** [comment]
We think this is likely a UI bug caused by CellBroadcastService having the same shared uid.

**Comment by thestinger** [comment]
This may be a UI bug caused by shared UID.

**Comment by xoraxax** [comment]
The CellBroadcastService lists as having versioncode 300000000 here, is that expected?

**Comment by thestinger** [comment]
Yes.

**Comment by xoraxax** [comment]
Also note that the "network manager" (not sure whether thats the original name, I am seeing the German name here) system app also has the same uid.

**Comment by muhomorr** [comment]
Can you capture a logcat after "Tethering" shows up in location indicator UI again?

**Comment by xoraxax** [comment]
`com.samsung.android.waterplugin` seems to be the culprit ... I forgot that the Samsung Watch 4 support is two fold: one general app and one specific app for the device class. I uninstalled only the general app last time I debugged this issue. Also deinstalling `com.samsung.android.waterplugin` stops the location info access by "Tethering". Not sure whether the underlying GOS bug is fixed, but knowing that this is only Samsung code trying to find its watch (which is not in range) is reassuring.

Here is the logcat output (captured from a GNU screen session leading to hard line breaks sometimes) - its trying to open a p2p connection:
[hardcopy.txt](https://github.com/GrapheneOS/os-issue-tracker/files/11415031/hardcopy.txt)

**Comment by Chippel** [comment]
Hey @xoraxax 
Seems like I have the same issue. I also use a Samsung smart watch. Could you tell me how to resolve this? I also wonder how I find the "com.samsung.android.waterplugin" thing. I'm a little bit lost here.

**Comment by xoraxax** [comment]
Hi @Chippel , the problem stopped for me after removing the smartwatch plugin. If this is not a viable solution for you, you need to wait for the display fix by the GrapheneOS team.

**Comment by Chippel** [comment]
@xoraxax do you mean the "Galaxy 5 manager" app?

**Comment by xoraxax** [comment]
Likely, the app has "plugin" in its name and refers to the model of your smartwatch. But note that you wont be able to use your smartwatch anymore after removing this watch model specific plugin app.

**Comment by fundusze** [comment]
Hi, I have the same issue. From time to time (hotspot is turned off) tethering shows green localization icon in the right up corner). I use Galaxy watch plugin to.
Is there a way to fix it without uninstalling it?
Pixel 7 with Androud 13, security update 05.08.2023

**Comment by flawedworld** [comment]
I suspect this is because the device would be scanning nearby Wi-Fi networks to determine an appropriate channel. This is a cosmetic issue, and not a bug.

**Comment by thestinger** [comment]
@xoraxax Do you still have this UI issue with current releases?

**Comment by xoraxax** [comment]
I do not have the apk installed anymore, @artproject-2004 can you maybe confirm?

**Comment by artproject-2004** [comment]
Hi, the issue still persists - from time to time green icon pop up in right corner. I have GW plugin installed (but it isn't used) and localization is still turn on.
